src/main/java/org/folio/circulation/domain/policy/RollingRenewalDueDateStrategy.java

```diff
@@ -80,10 +80,18 @@
   }
 
   private Result<DateTime> truncateDueDateBySchedule(
+    DateTime from,
     DateTime loanDate,
     DateTime dueDate) {
 
-    return dueDateLimitSchedules.truncateDueDate(dueDate, loanDate,
+    Result<DateTime> dueDateResult = findDueDateFor(loanDate, dueDate);
+    return dueDateResult.failed()
```

Comment:
What does this check represent?

My understanding of this change is to use a different (either current due or system) date rather than the loan date for these decisions.